Automattic/wp-calypso
# Performance Profiler: add message to insights if wpscan is still running

Related to https://github.com/Automattic/dotcom-forge/issues/9556

## Proposed Changes

This PR updates the query used for the performance insights to keep polling until both the `pagespeed` and `wpscan` reports are completed. Using the WPScan report status, it adds a message on top of the insights section if the WPScan tests are not done. In this state, the LLM recommendations have less context to work with and the suggestions might not be as accurate.

I experimented a bit with adding the message when the LLM suggestion is clicked, but hiding it behind an interaction seemed to be annoying UX-wise. When the WPScan results are done, the user would have to close then reopen each insight to see the LLM answer in this case.

I'm not 100% sure about the wording and the styling of the message to use here. I added an initial version, but some iterations/ideas on this would be very welcome!

## Why are these changes being made?


* We received feedback that on some sites, the WPScan process is running longer than the PSI one, and in these cases the LLM recommendations can't use the list of installed plugin for context.

## Testing Instructions



* Open the devtools Network panel
* Go to `/speed-test` and start a new test
* Check that the `insights` requests are still being polled after the `pagespeed` report is completed, and that the polling stops when the `wpscan` report is also completed
* Check the insights section. The message should be shown at the top of the section while the wpscan report is not completed.
* Check that the logged-in version works the same way.

| Page | WPScan queued/running | WPScan completed |
|--------|--------|--------|
| Logged-in | ![CleanShot 2024-11-26 at 14 05 58@2x](https://github.com/user-attachments/assets/09e708d4-5319-4968-aff9-9e52d5fba1e8) | ![CleanShot 2024-11-26 at 14 07 49@2x](https://github.com/user-attachments/assets/511840b6-056d-40d1-bf74-fd0d669b956e) |
| Logged-out | ![CleanShot 2024-11-26 at 14 23 51@2x](https://github.com/user-attachments/assets/db8d55fc-7d40-4f33-86ad-9b7d3c909ef0) | ![CleanShot 2024-11-26 at 14 23 13@2x](https://github.com/user-attachments/assets/8f2387fd-661a-4681-a980-ad683fbaafd1) |
